### DISCUSSION TITLE
Problems with Zim-wiki Flathub version for Xubuntu Lunar (23.04)

### DISCUSSION THREAD

**Discussion by Nick222** [zim-desktop-wiki/zim-desktop-wiki#2416] [Q&A]
What about Zim-wiki version for Ubuntu Lunar (23.04) ?

I cannot see it in main PPA...

**Comment by sojusnik** [comment]
Also realized that yesterday...

Maybe a good opportunity to test how Zim is running as Flatpak? The Flathub version is up to date: https://flathub.org/apps/org.zim_wiki.Zim

**Comment by Nick222** [comment]
That is terrible !

When You want to move Zim-wiki from PPA to Flatpack - You must:

At first - make a copy of Your notebooks...

Then:

1) Copy from **~/.config/zim/** to **~/.var/app/org.zim_wiki.Zim/config/zim/** files notebooks.list, preferences.conf, style.conf

2) Copy from **~/.config/gtk-3.0/** to **~/.var/app/org.zim_wiki.Zim/config/gtk-3.0/** file gtk.css

3) Copy from **~/.cache/zim/(Your_notebook)/** to **~/.var/app/org.zim_wiki.Zim/cache/zim/(Your_notebook)/** files index.db, state.conf

4) Copy from **~/.local/share/zim/plugins/** to **~/.var/app/org.zim_wiki.Zim/data/zim/plugins/** all plugins directories and files, then switch on (in Zim-wiki preferences) this plugins and configured it

Open Zim notebook in new version of program, manually reindex, save and close.

Then compare old version Your notebook - and new.

Best wishes... :)))

P.S.: And libzeitgeist cannot load and this plugin do not work (Xubuntu 23.04).

**Comment by Nick222** [comment]
Zim-wiki flatpack-version work slowly...

**Reply by sojusnik** [reply]
You mean that it's starting slowly or the general response time to opening notes etc. is slow?

**Reply by Nick222** [reply]
Past after copy or cut slowly

**Reply by sojusnik** [reply]
Do you observe this with other flatpak apps or only with Zim? Usually there shouldn't be any noticeable lag with flatpak apps AFAIK.

**Reply by Nick222** [reply]
After an hour of work, when I trying to insert characters anywhere, Zim starts to take 25 percent of the processor's resources, the cursor disappears (exactly in Zim) and the typed characters actually appear only after a few seconds.
This is not observed with any other program (including flatpack applications).

**Comment by Nick222** [comment]
When working with the Zim Clip extension (author Rui Nibau), for some reason this extension causes the wrong version of the program - the old one and with the old settings (and not installed through Flatpack with its settings).

Is this a bug in an extension or Zim-wiki program?

**Comment by Nick222** [comment]
And the user dictionary file is called not at all the same as before ...

It turns out that for all programs - one user dictionary, and for Zim-wiki - another (separate)?

**Comment by Nick222** [comment]
Do not work "Print to browser".

And do not load "libzeitgeist".

**Comment by Nick222** [comment]
About Zim Clip:

From plugin`s author :

> Zim-clip communicates with zim-wiki through native messaging hosts.
> Edit the file ~/.mozilla/native-messaging-hosts/**zimclip.json** and change the "path" property in this json file to target the flatpack installation.

Then I asked:

>  What must I write in **config.json** (path ~/.var/app/org.zim_wiki.Zim/data/zim/plugins/zimclip/)  instead ""path": "/app/bin/zim"" ? 

Plugin`s author  answered:

> You can try to edit it manually: "path" needs to point to the python script launching zim (zim.py). For example, on my desktop, "path" is "~/.local/bin/zim-desktop-wiki/zim.py" (i'm running zim from source).
> I don't know how flatpack works but "path" should be somethnig like "/path/to/where/flatpack/installed/zim/zim.py".

Can somebody say something about this "path" editing ?

**Comment by Nick222** [comment]
After change path to "/var/lib/flatpak/app/org.zim_wiki.Zim/x86_64/stable/active/files/bin/zim" - Zim Clip become work and Zim-wiki opened.

But it open old version of Zim-wiki (installed through PPA) and do not see my Zim Clip-for-Firefox options.

**Comment by Nick222** [comment]
I am:

>> I edit file config.json (in Zim Clip directory) and write new path to Zim - 
>> "/var/lib/flatpak/app/org.zim_wiki.Zim/x86_64/stable/active/files/bin/zim".
>> But after start Zim-wiki it again change to "/app/bin/zim"
>> What can I do to stop this revert changes ?  

Zim Clip plugin author:

> This value is written by the zimclip plugin reading a property given by 
> Zim itself (see ~/.local/share/zim/plugins/zimclip/__init__.py, line 119).
> I don't understand how you can run something from 
> "/var/lib/flatpak/app/org.zim_wiki.Zim/x86_64/stable/active/files/bin/zim" 
> and zim responding that it's executable is "/app/bin/zim"

**Comment by Nick222** [comment]
Can anybody answer me ?

**Reply by introt** [reply]
> I don't understand how you can run something from
"/var/lib/flatpak/app/org.zim_wiki.Zim/x86_64/stable/active/files/bin/zim"

That file is likely a copy of `zim.py` which imports "zim" from the global Python environment - eg. launches the "old version". To run the one in the flatpak (which sees itself as /app/bin/zim inside the sandbox) you likely need a script wrapper that runs the flatpak command.

Since you've already got a non-flatpak Zim installed, I'd suggest running Zim from the source code - it's all Python so nothing needs to be compiled. You can either use `git clone` or simply download and extract a source tarball. You can then point Zim Clip to the repository's `zim.py`.

**Reply by Nick222** [reply]
> Since you've already got a non-flatpak Zim installed

But I uninstall PPA-version Zim-wiki...

**Reply by Nick222** [reply]
And why then the flatpack version at all?

**Reply by introt** [reply]
> But I uninstall PPA-version Zim-wiki...

I mean that having it installed means you've got the dependencies already. You can install them manually, they're listed here: https://github.com/zim-desktop-wiki/zim-desktop-wiki#installing-from-source (or you can run `apt show zim | grep Depends`).

> And why then the flatpack version at all?

You don't need the flatpak version if you're running from source. If you're asking why it exists, you can see on flathub that it's not an official package; indeed, it's not listed under https://zim-wiki.org/downloads.html - this doesn't mean it isn't trustworthy, it just means that it's packaged independently from "us". It's good to have options for eg. the people running on immutable operating systems and/or without root access.

**Reply by Nick222** [reply]
It's not clear - now the PPA is not supported anymore?

**Reply by introt** [reply]
Ah, sorry, I misinterpreted the prior thread. According to https://repology.org/project/zim/versions Zim should be available for Xubuntu Lunar (23.04) and even Mantic (23.10) in the official/universe repositories. It seems the Launchpad PPA isn't currently configured to publish for Lunar. I've opened a ticket: #2428
